src/crepe/system/AISystem.cpp

```diff
@@ -0,0 +1,181 @@
+#include <algorithm>
+#include <cmath>
+
+#include "api/LoopTimer.h"
+#include "manager/ComponentManager.h"
+#include "manager/Mediator.h"
+
+#include "AISystem.h"
+
+using namespace crepe;
+
+void AISystem::update() {
+	const Mediator & mediator = this->mediator;
+	ComponentManager & mgr = mediator.component_manager;
+	RefVector<AI> ai_components = mgr.get_components_by_type<AI>();
+
+	//TODO: Use fixed loop dt (this is not available at master at the moment)
+	double dt = LoopTimer::get_instance().get_delta_time();
+
+	// Loop through all AI components
+	for (AI & ai : ai_components) {
+		if (!ai.active) {
+			continue;
+		}
+
+		RefVector<Rigidbody> rigidbodies
+			= mgr.get_components_by_id<Rigidbody>(ai.game_object_id);
+		if (rigidbodies.empty()) {
+			throw std::runtime_error(
+				"AI component must be attached to a GameObject with a Rigidbody component");
```

Comment:
Waarom throw je hier?
Is loggen niet beter?

Reply:
Ik wil hier, als er geen rigidbody is, stoppen met het uitvoeren van deze functie (want anders gaat het mis). Dit lijkt mij dus een geschikte plaatsen om te throwen.

Reply:
was even vergeten dat je nu al de rigidbody gebruikt ipv transform.